test/module/irohad/consensus/yac/yac_synchronization_test.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "module/irohad/consensus/yac/yac_fixture.hpp"
+
+#include "common/hexutils.hpp"
+
+using namespace iroha::consensus::yac;
+
+using ::testing::_;
+using ::testing::Return;
+
+/**
+ * The class helps to create fake network for unit testing of consensus
+ */
+class NetworkUtil {
+ public:
+  /// creates fake network of number_of_peers size
+  NetworkUtil(size_t number_of_peers) {
+    for (size_t i = 0; i < number_of_peers; ++i) {
+      peers_.push_back(makePeer(std::to_string(i)));
+    }
+    order_ = ClusterOrdering::create(peers_);
+  }
+
+  auto createHash(const iroha::consensus::Round &r,
+                  const std::string &block_hash = "default_block",
+                  const std::string &proposal_hash = "default_proposal") const {
+    return YacHash(r, proposal_hash, block_hash);
+  }
+
+  auto createVote(size_t from, const YacHash &yac_hash) const {
+    return iroha::consensus::yac::createVote(
+        yac_hash,
+        *iroha::hexstringToBytestring(peers_.at(from)->pubkey().hex()));
```

Comment:
Accessing vector elements by index looks scary, but can be tolerated in tests. Could you please add an assertion to catch indices outside the vector boundaries?